### PR TITLE
Updated cookie js

### PR DESCRIPTION
**Pull request by boxadesign**
Updated cookie components to conform with policy requirements. Changes:

- Sets all cookies to false by default (other than essential).
- Has the gov.uk style banner and secondary confirmation banner.
- Updated unit tests for new functionality.

### CODE REVIEW

File: src/tests/spec/cookies/cookies-banner.spec.js [hunk 1]
```diff
@@ -68,23 +94,33 @@ describe('Component: Cookie banner', function() {
...
   const formHTML =
     '<div class="cookies-banner">' +
-    '<div class="container">' +
-    '<div class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap@s">' +
-    '<div class="grid__col grid__col--flex col-auto u-flex-shrink@s">' +
-    '<p class="cookies-banner__desc u-mr-s">CENSUS.GOV.UK uses cookies to make the site simpler.</p>' +
-    '</div>' +
-    '<div class="grid__col grid__col--flex col-auto">' +
-    '<button type="button" class="btn btn--small js-accept-cookies">' +
+    '<div class="container cookies-banner__primary">' +
```
Comment (rmccar) [inline src/tests/spec/cookies/cookies-banner.spec.js:97]: Would be be better if this was indented

**Comment by boxadesign** [comment]
> After accepting cookie settings, the example on both the cookie banner and cookie settings pages is blank. I know this could be difficult to fix because technically thats what the component should do but it just looks like its broken to the user
> Screenshot 2020-06-10 at 11 57 42

We need it to do this so that it fully carries out all events. If it remains this will be misleading.

Also with the test indenting this is the correct way to lay out this type of code.

File: src/tests/spec/cookies/cookies-banner.spec.js (hunk 1 above)
Comment (boxadesign) [reply src/tests/spec/cookies/cookies-banner.spec.js:97]: That wouldn't be the correct way to do it

**Comment by rmccar** [comment]
Looks good apart from theres a huge gap on the cookie settings example between the end of the content and the footer

File: src/components/cookies-banner/_macro-options.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+| -------------------------- | ------ | -------- | ------------------------------------------------------------------------------------------------------------------ |
+| ariaLabel                  | string | false    | Aria label for the cookie banner region (default is 'cookies banner')                                              |
+| statementTitle             | string | true     | The title text for content for the cookie banner.                                                                  |
+| statementText              | string | true     | The text content for the cookie banner. HTML is allowable to embed links                                           |
+| confirmationText           | string | true     | The text content for the confirmation cookie message. HTML is allowable to embed links                             |
+| primaryButtonText          | string | false    | Text for the primary button (default is 'Accept cookies')                                                          |
+| secondaryButtonText        | string | false    | Text for the secondary button (default is 'Cookie settings')                                                       |
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro-options.md:8]:
```suggestion
| secondaryButtonText        | string | false    | Text for the secondary button (default is 'Set cookie preferences')                                                       |
```

File: src/patterns/cookies-settings/index.njk [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}}
+## How to use
+
+The preferences page uses groups of radio buttons which allow a user to allow or disallow cookies to be set. Each group of radios
+must contain a `name` attribute prefixed with `cookies-` following with `cookieType` name e.g. `cookies-campaigns`. This allows the 
+javascript to set the appropriate cookies. 
+
+If javascript is disabled a message is displayed explaining that javascript is required to set and change cookies.
+
+When the form is submitted a banner is shown to tell the user that preferences have been updated. The form is handled by javascript.
```
Comment (jrbarnes9) [inline src/patterns/cookies-settings/index.njk:49]:
```suggestion
When the form is submitted a success panel is shown to tell the user that their preferences have been updated. The form is handled by javascript.
```
Where is the contents of this panel set?

**Comment by boxadesign** [comment]
> Looks good apart from theres a huge gap on the cookie settings example between the end of the content and the footer

Yeah slightly unavoidable due to the way the responsive iframe works.

File: src/patterns/cookies-settings/index.njk (hunk 3 above)
Comment (boxadesign) [reply src/patterns/cookies-settings/index.njk]: All content is within the template for this page. It's not a macro due to the free form nature of the content. I did consider making it a macro but it's not really feasible. It's just a page of components so guidance is what we need to go with.

**Comment by rmccar** [comment]
> > Looks good apart from theres a huge gap on the cookie settings example between the end of the content and the footer
> 
> Yeah slightly unavoidable due to the way the responsive iframe works.

Ahh thats annoying

**Comment by jrbarnes9** [comment]
> Looks good apart from theres a huge gap on the cookie settings example between the end of the content and the footer

iframe resizer not working?

File: src/components/cookies-banner/examples/cookies-banner/index.njk [hunk 4]
```diff
@@ -4,7 +4,9 @@
 {% from "components/cookies-banner/_macro.njk" import onsCookiesBanner %}
 {{
     onsCookiesBanner({
-        "html": 'We use cookies to improve your experience of census.gov.uk',
+        "statementTitle": 'Tell us whether you accept cookies',
+        "statementText": 'We use You can <a href="#">cookies to collect information</a> about how you use census.gov.uk. We use this information to make the website work as well as possible and improve our services.',
```
Comment (rmccar) [inline src/components/cookies-banner/examples/cookies-banner/index.njk:8]:
```suggestion
        "statementText": 'We use <a href="#">cookies to collect information</a> about how you use census.gov.uk. We use this information to make the website work as well as possible and improve our services.',
```

**Pull request merged**